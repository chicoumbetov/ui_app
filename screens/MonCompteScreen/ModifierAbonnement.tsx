import React, { useEffect } from 'react';
import {
  StyleSheet, View,
} from 'react-native';
import {
  Button, Layout, Text,
} from '@ui-kitten/components';
import { useNavigation } from '@react-navigation/native';
import TextInputComp from '../../components/Form/TextInput';
import MaxWidthContainer from '../../components/MaxWidthContainer';

const ModifierAbonnement = () => {
  const navigation = useNavigation();
  const onPress = () => {
    navigation.navigate('ModifierInfo2');
  };

  useEffect(() => {
    console.log('useEffect test of ModifierInfo 1');
  });

  return (

    <MaxWidthContainer
      withScrollView="keyboardAware"
      outerViewProps={{
        style: {
          backgroundColor: '#efefef',
        },
        showsVerticalScrollIndicator: false,
      }}
      innerViewProps={{
        style: {
          paddingHorizontal: 24,
          paddingVertical: 34,
        },
      }}
    >
      <View>
        <Text category="h1" style={styles.title}>Modifier vos abonnements</Text>
      </View>

      <TextInputComp
        name="prenom"
        placeholder="Changer prÃ¨nom"
      />

      <Layout style={{ marginTop: 20, backgroundColor: 'transparent' }}>
        <TextInputComp
          name="nom"
          placeholder="Changer nom"
        />
      </Layout>

      <View style={styles.buttonRight}>
        <Button onPress={onPress} size="large" style={{ width: 139 }}>
          Valider
        </Button>
      </View>

    </MaxWidthContainer>

  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 21,
    backgroundColor: 'transparent',
  },
  buttonRight: { alignItems: 'flex-end', marginTop: 34 },
  title: {
    marginTop: 12,
    marginBottom: 39,
  },
});

export default ModifierAbonnement;
