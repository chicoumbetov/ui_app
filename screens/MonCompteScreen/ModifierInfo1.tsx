import React, { useEffect } from 'react';
import {
  StyleSheet, View,
} from 'react-native';
import {
  Button, Text,
} from '@ui-kitten/components';
import { useNavigation } from '@react-navigation/native';
import { useForm } from 'react-hook-form';
import TextInput from '../../components/Form/TextInput';
import MaxWidthContainer from '../../components/MaxWidthContainer';
import { AvailableValidationRules } from '../../components/Form/validation';
import Radio from '../../components/Form/Radio';
import PhoneNumberInput from '../../components/Form/PhoneNumberInput';
import Form from '../../components/Form/Form';
import useCurrentUser from '../../src/API/User';
import { useAuth } from '../../utils/CustomHooks';

type ModifierInfo1Form = {
  firstname:string;
  lastname: string;
  email: string;
  password: string;
  phoneNumber: string;
  optIn: boolean;
};

const ModifierInfo1 = () => {
  const navigation = useNavigation();
  const { user: authUser } = useAuth();
  const { updateUser, user } = useCurrentUser();

  const onPress = async (data: ModifierInfo1Form) => {
    const { password, ...otherProps } = data;

    if (user && authUser) {
      await updateUser({
        variables: {
          input: {
            id: user.id,
            ...otherProps,
            // eslint-disable-next-line no-underscore-dangle
            _version: user._version,
          },
        },
      });

      navigation.navigate('modifier-info-2');
    }
  };

  const modifierInfo1Form = useForm<ModifierInfo1Form>();

  return (

    <MaxWidthContainer
      withScrollView="keyboardAware"
      outerViewProps={{
        style: {
          backgroundColor: '#efefef',
        },
        showsVerticalScrollIndicator: false,
      }}
      innerViewProps={{
        style: {
          paddingHorizontal: 24,
          paddingVertical: 34,
        },
      }}
    >
      <View>
        <Text category="h1" style={styles.title}>Modifier vos informations</Text>
      </View>
      <Form<ModifierInfo1Form> {...modifierInfo1Form}>
        <>
          <TextInput
            name="firstname"
            placeholder="Votre prénom"
            validators={[
              AvailableValidationRules.required,
            ]}
            defaultValue={user?.firstname || undefined}
          />

          <TextInput
            name="lastname"
            placeholder="Votre nom"
            containerStyle={{ marginTop: 20 }}
            validators={[
              AvailableValidationRules.required,
            ]}
            defaultValue={user?.lastname || undefined}
          />

          <TextInput
            name="email"
            placeholder="Votre  e-mail"
            containerStyle={{ marginTop: 20 }}
            validators={[
              AvailableValidationRules.required,
            ]}
            defaultValue={user?.email || undefined}
          />

          <TextInput
            name="password"
            placeholder="Votre mot de passe"
            secureTextEntry
            withEyeToggle
            containerStyle={{ marginTop: 20 }}
          />

          <PhoneNumberInput
            name="phoneNumber"
            placeholder="Votre numéro de téléphone"
            containerStyle={{ marginTop: 20 }}
            validators={[
              AvailableValidationRules.numeroTel,
              AvailableValidationRules.required,
            ]}
            defaultValue={user?.phoneNumber || undefined}
          />
          <Radio
            name="optIn"
            label="Souhaitez-vous rester informé de nos actualités ? "
            labelPosition="before"
            style={{ marginTop: 20 }}
            value={user?.optIn || false}
          />

          <View style={styles.buttonRight}>
            <Button onPress={modifierInfo1Form.handleSubmit((data) => onPress(data))} disabled={user === null} size="large" style={{ width: 139 }}>
              Valider
            </Button>
          </View>
        </>
      </Form>

    </MaxWidthContainer>

  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 21,
    backgroundColor: 'transparent',
  },
  buttonRight: { alignItems: 'flex-end', marginTop: 34 },
  title: {
    marginTop: 12,
    marginBottom: 39,
  },
});

export default ModifierInfo1;
