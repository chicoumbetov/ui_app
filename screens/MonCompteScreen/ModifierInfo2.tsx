import React, { useEffect } from 'react';
import { ScrollView, StyleSheet, View } from 'react-native';
import {
  Button, Layout, Text,
} from '@ui-kitten/components';
import { useNavigation } from '@react-navigation/native';
import TextInputComp from '../../components/Form/TextInput';

const ModifierInfo2 = () => {
  const navigation = useNavigation();

  const onPress = () => {
    navigation.navigate('ModifierInfo3');
  };

  useEffect(() => {
    console.log('useEffect test of ModifierInfo 2');
  });

  return (
    <ScrollView
      showsVerticalScrollIndicator={false}
      style={styles.container}
    >
      <View>
        <Text category="h1" style={styles.title}>Modifier vos informations</Text>
      </View>

      <View style={{
        flexDirection: 'row', marginBottom: 20, alignItems: 'center',
      }}
      >
        <View style={{ marginRight: 20 }}>
          <Text category="p1">Votre date de naissance</Text>
        </View>

        <Layout style={{ flex: 1, backgroundColor: 'transparent' }}>
          <TextInputComp
            name="dateDeNaissance"
            placeholder="dd/mm/yyyy"
            icon="calendar-outline"
          />
        </Layout>
      </View>

      <TextInputComp
        name="adresse"
        placeholder="Adresse"
      />

      <Layout style={{ marginTop: 20, backgroundColor: 'transparent' }}>
        <TextInputComp
          name="adresseComplement"
          placeholder="ComplÃ©ment d'adresse"
        />
      </Layout>

      <Layout style={{ marginTop: 20, backgroundColor: 'transparent' }}>
        <TextInputComp
          name="codePostal"
          placeholder="Code postale"
        />
      </Layout>

      <Layout style={{ marginTop: 20, backgroundColor: 'transparent' }}>
        <TextInputComp
          name="ville"
          placeholder="Ville"
        />
      </Layout>

      <Layout style={{ marginTop: 20, backgroundColor: 'transparent' }}>
        <TextInputComp
          name="pays"
          placeholder="Pays"
        />
      </Layout>

      <View style={styles.buttonRight}>
        <Button onPress={onPress} size="large" style={{ width: 139 }}>
          Valider
        </Button>
      </View>

    </ScrollView>

  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 21,
    backgroundColor: 'transparent',
  },
  buttonRight: { alignItems: 'flex-end', marginTop: 34 },
  title: {
    marginTop: 12,
    marginBottom: 39,
  },
});

export default ModifierInfo2;
