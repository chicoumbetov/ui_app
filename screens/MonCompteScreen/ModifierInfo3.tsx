import React, { useState } from 'react';
import {
  StyleSheet, TouchableOpacity, View,
} from 'react-native';
import { Button, Layout, Text } from '@ui-kitten/components';
import { useLinkTo, useNavigation } from '@react-navigation/native';
import * as ImagePicker from 'expo-image-picker';
import MaxWidthContainer from '../../components/MaxWidthContainer';

import ManAvatar from '../../assets/Omedom_Icons_svg/Avatars/manAvatar.svg';
import WomanAvatar from '../../assets/Omedom_Icons_svg/Avatars/womanAvatar.svg';

const Informations = () => {
  const [value, setValue] = React.useState('');
  const [avatarImage, setAvatarImage] = useState('MaisonVerte');

  const navigation = useNavigation();
  const linkTo = useLinkTo();

  const onPress = () => {
    linkTo('/');
  };

  const onTakePicture = () => (navigation.navigate('CameraDom'));

  const pickImage = async () => {
    try {
      const result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.Images,
      });
    } catch (e) {
      console.log('pickImage error: ', e);
    }
  };

  // Avatar changement
  let SelectedAvatar = ManAvatar;
  switch (avatarImage) {
    case 'ManAvatar':
      SelectedAvatar = ManAvatar;
      break;
    case 'WomanAvatar':
      SelectedAvatar = WomanAvatar;
      break;
  }

  return (
    <MaxWidthContainer outerViewProps={{
      style: {
        backgroundColor: '#efefef',
        paddingHorizontal: 24,
        paddingVertical: 34,
      },
    }}
    >

      <Text category="h1" style={styles.title}>Modifier vos informations</Text>
      <Text category="h2">Changer votre photo de profil</Text>
      <Layout style={{
        alignItems: 'center', backgroundColor: 'transparent', marginVertical: 45, marginTop: 8,
      }}
      >
        <SelectedAvatar height={140} width={140} />
      </Layout>

      <TouchableOpacity onPress={() => {}}>
        <Text category="h5" appearance="hint">Choisir une icone</Text>
      </TouchableOpacity>

      <Layout style={{
        flexDirection: 'row', marginTop: 21, justifyContent: 'space-around', alignItems: 'center', backgroundColor: 'transparent',
      }}
      >
        <TouchableOpacity onPress={() => { setAvatarImage('ManAvatar'); }}>
          <ManAvatar height={50} width={50} />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => { setAvatarImage('WomanAvatar'); }}>
          <WomanAvatar height={50} width={50} />
        </TouchableOpacity>

      </Layout>

      <TouchableOpacity onPress={() => { onTakePicture(); }} style={{ marginVertical: 39 }}>
        <Text category="h5" status="info">Prendre une photo</Text>
      </TouchableOpacity>

      <View style={{ flexDirection: 'row', justifyContent: 'space-between', marginVertical: 1 }}>
        <TouchableOpacity onPress={() => { pickImage(); }}>
          <Text category="h5" status="info">Ajouter une photo</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {}}>
          <Text category="h5" status="basic">Supprimer la photo</Text>
        </TouchableOpacity>
      </View>

      <View style={{
        flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', marginVertical: 36,
      }}
      >
        <TouchableOpacity onPress={() => {}}>
          <Text category="h5" status="basic">Ignorer</Text>
        </TouchableOpacity>
        <Button onPress={onPress} size="large">
          Valider
        </Button>
      </View>

    </MaxWidthContainer>

  );
};

const styles = StyleSheet.create({
  container: {
    margin: 22,
    backgroundColor: 'transparent',
  },
  title: {
    marginTop: 18,
    marginBottom: 27,
  },
  button: {
    fontSize: 17,
    color: '#0076c8',
  },
  buttonRight: {
    alignItems: 'flex-end',
  },
});

export default Informations;
