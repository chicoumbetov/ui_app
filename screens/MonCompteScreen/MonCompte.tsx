/**
 * Mon Compte to visualise compte info or modify.
 *
 * @author: Shynggys UMBETOV
 */
import * as React from 'react';
import { Text } from '@ui-kitten/components';

import { View } from 'react-native';
import Informations from './Components/Informations';
import Abonnement from './Components/Abonnement';
import MaxWidthContainer from '../../components/MaxWidthContainer';

import { useUser } from '../../src/API/UserContext';
import AutoAvatar from '../../components/AutoAvatar';
import Separator from '../../components/Separator';

export default function MonCompte() {
  const { user } = useUser();

  return (
    <MaxWidthContainer
      withScrollView="keyboardAware"
      outerViewProps={{
        showsVerticalScrollIndicator: false,
        style: {
          paddingHorizontal: 46, paddingTop: 37,
        },
      }}
    >

      <Text category="h1">
        Mon Compte
      </Text>
      <View style={{
        alignItems: 'center', backgroundColor: 'transparent', marginTop: 41, marginBottom: 34,
      }}
      >
        <AutoAvatar
          style={{
            height: 108, width: 108, borderRadius: 54, overflow: 'hidden',
          }}
          avatarInfo={user?.avatarUri || 'default::ManAvatar'}
        />
        <Text category="h2" style={{ marginTop: 15 }}>
          {user?.firstname || undefined}
        </Text>
      </View>
      <Separator />

      <Informations utilisateur={user} />
      <Separator />
      <Abonnement />

    </MaxWidthContainer>
  );
}
