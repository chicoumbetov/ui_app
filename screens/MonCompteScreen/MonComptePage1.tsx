/**
 * Mon Compte to visualise compte info or modify.
 *
 * @author: Shynggys UMBETOV
 */

import * as React from 'react';

import { Layout, Text } from '@ui-kitten/components';
import {
  Image, ScrollView, StyleSheet,
} from 'react-native';
// import { API, graphqlOperation, DataStore } from 'aws-amplify';
import { useState } from 'react';
import Informations from './Components/Informations';

// import { Client } from '../../src/models';
// import { getClient } from '../../src/graphql/queries';

import clientData from '../../mockData/clientDATA';
import Abonnement from './Components/Abonnement';

export default function MonComptePage1() {
  const [clients, updateClients] = useState(clientData);

  // console.log('clients3:', clients);

  /*
    // DataStore connexion
    async function fetchClient() {
      try {
        const clients = await DataStore.query(Client);
        console.log('clients: ', clients);
      } catch (e) {
        console.log('Error retrieving posts', e);
      }
    }

    useEffect(() => {
      fetchClient();
      const subscription = DataStore.observe(Client).subscribe(() => getClient);
      return () => subscription.unsubscribe();
    });

    useEffect(() => {
      const fetchClient2 = async () => {
        try {
          const clientData = await API.graphql(
            graphqlOperation(
              getClient, {
                id: '4a62b89d-b6ab-48ee-b84e-d1b140e959c9',
              },
            ),
          );
          console.log('clientData: ', clientData);
        } catch (e) {
          console.log(e);
        }
      };
      fetchClient2();
    }, []);

   */

  return (
    <Layout style={styles.container}>
      <ScrollView
        showsVerticalScrollIndicator={false}
      >

        <Layout style={{
          flex: 1, backgroundColor: 'rgba(246, 246, 246, 0.5)', paddingHorizontal: 22, paddingTop: 37,
        }}
        >
          <Text category="h1">
            Mon Compte
          </Text>
          <Layout style={{
            alignItems: 'center', backgroundColor: 'rgba(246, 246, 246, 0.5)', marginTop: 41, marginBottom: 34,
          }}
          >
            <Image
              source={require('../../assets/Icones_omedom/avatars/avatar_1.png')}
              style={{ height: 109, width: 108 }}
            />

            <Text style={{ fontSize: 22, fontWeight: '400', marginTop: 15 }}>
              {clients.Client.fields[0].nom}
            </Text>
          </Layout>
        </Layout>

        <Informations clientData={clients} />
        <Abonnement clientData={clients} />

      </ScrollView>
    </Layout>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1, backgroundColor: '#efefef',
  },
});
