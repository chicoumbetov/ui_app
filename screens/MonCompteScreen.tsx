import * as React from 'react';

import { Layout } from '@ui-kitten/components';
import MonCompte from '../components/MonCompte/MonCompte';

export default function MonCompteScreen() {
  return (
    <Layout style={{ flex: 1, backgroundColor: 'rgba(246, 246, 246, 0.5)' }}>
      <MonCompte />
    </Layout>
  );
}
