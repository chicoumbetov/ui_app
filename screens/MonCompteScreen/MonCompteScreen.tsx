/**
 * Mon Compte to visualise compte info or modify.
 *
 * @author: Shynggys UMBETOV
 */

import * as React from 'react';

import { Layout } from '@ui-kitten/components';
import { createStackNavigator } from '@react-navigation/stack';
import { DrawerActions } from '@react-navigation/native';
import { TouchableOpacity } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import MonComptePage1 from './Components/MonComptePage1';
import ModifierInfo1 from './Components/ModifierInfo1';
import ModifierInfo2 from './Components/ModifierInfo2';
import ModifierInfo3 from './Components/ModifierInfo3';
import HeaderLeftOpenDrawerNavigation from '../../navigation/HeaderLeftOpenDrawerNavigation';
import HeaderRightOpenDrawerNavigation from '../../navigation/HeaderRightOpenDrawerNavigation';
import LogoPicture from '../../components/LogoPicture/LogoPicture';
import CameraDom from './Components/Camera';

const Stack = createStackNavigator();

export default function MonCompteScreen({ navigation }) {
  return (
    <Layout style={{ flex: 1, backgroundColor: '#efefef' }}>
      <Stack.Navigator
        initialRouteName="Mon Compte"
      >
        <Stack.Screen
          name="MonCompte"
          options={{
            headerTitle: false,
            headerLeftContainerStyle: {
              paddingBottom: 5,
            },
            headerRightContainerStyle: {
              marginBottom: 5,
            },
            headerLeft: () => (
              <TouchableOpacity
                onPress={() => {
                  navigation.dispatch(DrawerActions.toggleDrawer());
                }}
              >
                <AntDesign name="arrowleft" size={31} style={{ color: '#b5b5b5', marginLeft: 20 }} />
              </TouchableOpacity>

            ),
            headerRight: () => (
              <HeaderLeftOpenDrawerNavigation navigation={navigation} />
            ),
          }}
          component={MonComptePage1}
        />
        <Stack.Screen
          name="ModifierInfo1"
          options={{
            headerTitle: false,
            headerLeftContainerStyle: {
              paddingBottom: 5,
            },
            headerRightContainerStyle: {
              marginBottom: 5,
            },
            headerLeft: () => (
              <TouchableOpacity onPress={() => { navigation.navigate('MonCompte'); }}>
                <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
              </TouchableOpacity>

            ),
            headerRight: () => (
              <HeaderLeftOpenDrawerNavigation navigation={navigation} />
            ),
          }}
          component={ModifierInfo1}
        />
        <Stack.Screen
          name="ModifierInfo2"
          options={{
            headerTitle: false,
            headerLeftContainerStyle: {
              paddingBottom: 5,
            },
            headerRightContainerStyle: {
              marginBottom: 5,
            },
            headerLeft: () => (
              <TouchableOpacity onPress={() => { navigation.navigate('ModifierInfo1'); }}>
                <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
              </TouchableOpacity>

            ),
            headerRight: () => (
              <HeaderLeftOpenDrawerNavigation navigation={navigation} />
            ),
          }}
          component={ModifierInfo2}
        />
        <Stack.Screen
          name="ModifierInfo3"
          options={{
            headerTitle: false,
            headerLeftContainerStyle: {
              paddingBottom: 5,
            },
            headerRightContainerStyle: {
              marginBottom: 5,
            },
            headerLeft: () => (
              <TouchableOpacity onPress={() => { navigation.navigate('ModifierInfo2'); }}>
                <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
              </TouchableOpacity>

            ),
            headerRight: () => (
              <HeaderLeftOpenDrawerNavigation navigation={navigation} />
            ),
          }}
          component={ModifierInfo3}
        />
        <Stack.Screen
          name="CameraDom"
          options={{
            headerTitle: false,
            headerLeft: () => (
              <TouchableOpacity onPress={() => { navigation.navigate('ModifierInfo3'); }}>
                <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
              </TouchableOpacity>

            ),
          }}
          component={CameraDom}
        />
      </Stack.Navigator>
    </Layout>
  );
}
