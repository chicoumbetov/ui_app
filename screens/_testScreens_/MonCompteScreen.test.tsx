import renderer from 'react-test-renderer';
import MonCompteScreen from '../MonCompteScreen';

it('renders TabContactScreen', () => {
  expect(MonCompteScreen);
});

describe('<MonCompteScreen />', () => {
  it('renders correctly across screens', () => {
    const tree = renderer
      .create('<MonCompteScreen />').toJSON();
    expect(tree).toMatchSnapshot();
  });
});
