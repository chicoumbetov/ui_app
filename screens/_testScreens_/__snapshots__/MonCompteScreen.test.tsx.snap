// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonCompteScreen /> renders correctly across screens 1`] = `"<MonCompteScreen />"`;
